src/DependencyInjection/CsaGuzzleExtension.php

```diff
@@ -122,6 +124,11 @@ private function processCacheConfiguration(array $config, ContainerBuilder $cont
 
     private function processClientsConfiguration(array $config, ContainerBuilder $container, $debug)
     {
+        if (empty($config['default_client'])) {
+            $keys = array_keys($config['clients']);
```

Comment:
This looks like it will set the default_client as the first client if it is not explicitly stated. This feels ripe for potential unexpected misconfiguration - automatically injecting the wrong client and people not noticing. From what I've saw of Symfony's ideology they aim to make sure that there is no "magic" and that all configuration must be opted into.vAs a counterpoint to that they've chosen to [automatically register aliases for interfaces when there is only a single implementation](https://github.com/symfony/symfony/pull/25282), suggesting that automatic configuration is acceptable when it is unambiguous.

Thus I wonder if this automatic registration should be removed (the explict opt-into everything approach) or only be  triggered when there is only a single client defined (i.e. changing this if to  `if (empty($config['default_client']) && count($config['clients'] === 1)`). My gut feel would be for the latter.

Reply:
Also should there be some error handling around if a user enters an invalid value in default_client field? It would probably throw something like a "reference to undefined service" error if left alone but the bundle should be able to give throw a more useful  "tried to set default_client to an unknown client" error.

Reply:
Actually, that approach has directly been taken from doctrine bundle’s DI extension. I can make sure that we only set a default alias in the case when there is only a single service defined, this is also fine by me.

Reply:
Regarding error handling, this is something which can not be done as services may actually be defined outside from the client configuration.

This means that currently, only services defined in the semantic config can be aliased.

Reply:
If it's good enough for Doctrine then it's good enough for me :)  It looks like they do "set the default to the first one defined if you don't explicitly state it" thing too (see the bottom of https://symfony.com/doc/master/bundles/DoctrineBundle/configuration.html, the class name based service is an alias to database_connection), so leaving this as-is sounds like a good plan. My PR strat tends to be "lets follow other well-known stuff" so people can have an "ooh it's like that other thing" moment rather than any particularly strong opinions of my own. :)

---

> Regarding error handling, this is something which can not be done as services may actually be defined outside from the client configuration. This means that currently, only services defined in the semantic config can be aliased.

I think that's fine. If people go down the route of defining tagged services rather than using configuration then if they want a client created as a tagged service to be the default then they can easily add an alias for Client/ClientInterface within their own service config.

That said, Doctrine doesn't bother with any validation either so I reckon it's fine to leave this as-is too :)

---

(For future reference, this is the doctrine config (https://github.com/doctrine/DoctrineBundle/blob/4ab04c00f1c6449c4aaa3fc60d57c3238ea3530e/DependencyInjection/DoctrineExtension.php#L96-L99)

Reply:
I have provided additional information in the semantic configuration. I'll merge this in the upcoming days, if there's no remaining argument against this :)

Thanks for the feedback by the way, @BPScott! :)